src/dams_mcda/WSM.R

```diff
@@ -40,190 +45,386 @@ available_dams <- seq(1:8)
 # list of alternatives
 available_alternatives <- seq(1:5)
 
-matrix_cols <- length(criteria_inputs) # 14 default (output size, adds summedscore)
+matrix_cols <- length(criteria_inputs) # 14 default (output size)
 matrix_rows <- length(available_dams) # 8 default
 matrix_levs <- length(available_alternatives)
 
 
-WSM <- function(RawCriteriaMatrix, DamsData){
+WSM <- function(RawCriteriaMatrix, DamsDataMatrix, Ind_DamsDataMatrix){
 
 	# matrix setup
-	matrix_cols <- length(criteria_inputs) # 14 default (output size, adds summedscore)
+	matrix_cols <- length(criteria_inputs) # 14 default (output size)
 	matrix_rows <- length(available_dams) # 8 default
-	matrix_levs <- length(available_alternatives)
+	matrix_levs_ind <- length(available_alternatives)# 5 default
+	matrix_levs <- length(1:995)

```

Comment:
why so may levels

Reply:
also there is two occurrences of these variables

Reply:
meant to delete the old matrix_levs <- length(available_alternatives) to replace with matrix_levs<- length(1:995). The reason for so many levels is because Sam's f_raw dataset has 995 scenarios (i.e. unique permutations of possible multi-dam decisions given the set of 8 dams and 5 decision alternatives). For the individual dam decisions, the set of alternatives is still 5 (3rd dim of most of our 3D matrices), but we need to expand it to handle all of the 'scenario' possibilities for the multi-dam (combined) results.